CMakeLists.txt

```diff
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 3.23)
 project(BattleCity)
 
-set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD 23)
 
-add_executable(BattleCity main.cpp)
+add_executable(BattleCity message/FieldMessage.cpp main.cpp
+        message/FieldMessage.h message/FieldMessage.cpp
+        message/FieldMessageSerializer.h message/FieldMessageSerializer.cpp)
```

Comment:
To simplify testing, it is better to extract these files into new cmake library target. 
Like this: 
```cmake
# is only for cmake
add_library(${PROJECT_NAME}_INCLUDE INTERFACE)
target_include_directories(${PROJECT_NAME}_INCLUDE 
        INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/include)

# will compiled to dynamic or static library
add_library(${PROJECT_NAME}_LIB [STATIC | SHARED])
target_link_libraries(${PROJECT_NAME}_LIB PRIVATE 
        ${PROJECT_NAME}_INCLUDE)
target_sources(${PROJECT_NAME}_LIB PRIVATE 
        [your sources]...)

add_executable(${PROJECT_NAME}
        ${CMAKE_CURRENT_SOURCE_DIR}/src/main.cpp)
target_link_libraries(${PROJECT_NAME} PRIVATE 
        ${PROJECT_NAME}_LIB)
```